packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -195,14 +202,103 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 		return entry.public;
 	}
 
-	private localUpdate(data: DatastoreMessageContent, _forceBroadcast: boolean): void {
-		const content = {
-			sendTimestamp: Date.now(),
+	private queuedMessage: DatastoreUpdateMessage["content"] | undefined;
+
+	/**
+	 * The time at which the presence data must be sent. When presence updates are submitted, this value is calculated
+	 * based on the allowable latency for the update. A timer will fire at this time, sending any queued message
+	 * immediately. Note that if updates come in with allowable latencies that require sending before this deadline, those
+	 * messages will be sent immediately. Therefore this value can be considered the time of the next scheduled presence
+	 * update.
+	 */
+	private sendMessageDeadline: number = 0;
+
+	private localUpdate(data: DatastoreMessageContent, options: LocalUpdateOptions): void {
+		const allowableUpdateLatency = options.allowableUpdateLatency ?? 0;
+		const forceBroadcast = options.forceBroadcast;
+
+		const now = Date.now();
+		const updateDeadline = now + allowableUpdateLatency;
+		if (this.queuedMessage === undefined) {
+			// no queued message, so set the deadline based on the current message's allowable latency
+			this.sendMessageDeadline = updateDeadline;
+		}
+
+		// This function-local "datastore" will hold the merged message data
+		const queueDatastore: {
+			[WorkspaceAddress: string]: ValueElementMap<PresenceStatesSchema>;
+		} = this.queuedMessage?.data ?? {};
+
+		const currentMessageData: DatastoreMessageContent = data;
+
+		// Iterate over the current message data; individual items are workspaces
+		for (const workspaceName of Object.keys(currentMessageData)) {
+			const workspaceData = currentMessageData[workspaceName];
+
+			// Initialize the merged data as the queued datastore entry for the workspace
+			// Since the key might not exist, create an empty object in that case. It will
+			// be set explicitly after the loop.
+			const mergedData = queueDatastore?.[workspaceName] ?? {};
+
```

Comment:
Doesn't look like queueDatastore could be undefined here.
```suggestion
   const mergedData = queueDatastore[workspaceName] ?? {};
```